### PR TITLE
Beta WDLs can be pinned to a version [VS-1032]

### PR DESCRIPTION
**Pull request by mcovarr**

### CODE REVIEW

**Review by rsasch — APPROVED** [review]

File: scripts/variantstore/wdl/GvsUtils.wdl [hunk 1]
```diff
@@ -1,18 +1,40 @@
...
-    docker: "ubuntu:22.04"
+    docker: cloud_sdk_docker_decl
   }
   output {
     String basic_docker = "ubuntu:22.04"
-    String cloud_sdk_docker = "gcr.io/google.com/cloudsdktool/cloud-sdk:435.0.0-alpine"
+    String cloud_sdk_docker = cloud_sdk_docker_decl # Defined above as a declaration.
     # GVS generally uses the smallest `alpine` version of the Google Cloud SDK as it suffices for most tasks, but
     # there are a handlful of tasks that require the larger GNU libc-based `slim`.
     String cloud_sdk_slim_docker = "gcr.io/google.com/cloudsdktool/cloud-sdk:435.0.0-slim"
```
[the remark this answers is not in the document]
Comment (mcovarr) [reply scripts/variantstore/wdl/GvsUtils.wdl:59]: All the dockers are inputs which are captured in terra, exact digests in the Cromwell logs.

File: scripts/variantstore/wdl/GvsUtils.wdl [hunk 2]
```diff
@@ -1,18 +1,40 @@
...
+
+    bash ~{monitoring_script} > monitoring.log &
+
+    # install git
+    apk update && apk upgrade
+    apk add git
+
+    # The `--branch` parameter to `git clone` actually does work for tags which works out well for beta.
+    # https://git-scm.com/docs/git-clone#Documentation/git-clone.txt--bltnamegt
+    git clone https://github.com/broadinstitute/gatk.git --depth 1 --branch ~{git_branch_or_tag} --single-branch
```
[the remark this answers is not in the document]
Comment (mcovarr) [reply scripts/variantstore/wdl/GvsUtils.wdl]:
```
% git clone https://github.com/broadinstitute/gatk.git --depth 1 --branch not_a_real_branch
Cloning into 'gatk'...
warning: Could not find remote branch not_a_real_branch to clone.
fatal: Remote branch not_a_real_branch not found in upstream origin
```

**Review by RoriCremer — APPROVED** [review]
I kind of wish there was a diagram for all this haha

**Review by gbggrant — APPROVED** [review]

File: scripts/variantstore/variant_annotations_table/GvsCreateVATFilesFromBigQuery.wdl [hunk 3]
```diff
@@ -16,11 +18,15 @@ workflow GvsCreateVATFilesFromBigQuery {
...
-        call Utils.GetToolVersions
+    if (!defined(git_hash) || !defined(cloud_sdk_docker)) {
+        call Utils.GetToolVersions {
+            input:
+                git_branch_or_tag = git_branch_or_tag,
+        }
     }
 
     String effective_cloud_sdk_docker = select_first([cloud_sdk_docker, GetToolVersions.cloud_sdk_docker])
+    String effective_git_hash = select_first([git_hash, GetToolVersions.git_hash])
```
[the remark this answers is not in the document]
Comment (mcovarr) [reply scripts/variantstore/variant_annotations_table/GvsCreateVATFilesFromBigQuery.wdl:29]: `GetToolVersions.git_hash` is a non-optional String so no tragedies will ensue. If the git hash was not fetched this output will have the literal value `undefined`.

**Pull request merged**